### PR TITLE
Dataloader types

### PR DESCRIPTION
**Pull request by chrisvanmook**

### CODE REVIEW

File: src/fixtures/openapi/types.ts [hunk 1]
```diff
@@ -0,0 +1,231 @@
+export interface Prop {
```
Comment (jonkoops) [inline src/fixtures/openapi/types.ts:1]: This is where that generic API client would be great to have :100:

File: src/generated/drupal.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
 export const DRUPAL_THEME_FILTER_IDS = new Map([
```
Comment (jonkoops) [inline src/generated/drupal.ts:1]: Why have these ids suddenly changed?

File: src/graphql/config.ts [hunk 3]
```diff
@@ -1,17 +1,13 @@
...
-    data: {
-      load: Function
-      clear: Function
-    }
-    datasets: {
-      load: Function
-      clear: Function
-    }
+    cms: DataLoader<string, PromiseSettledResult<JsonAPI>, string>
+    data: DataLoader<string, PromiseSettledResult<{ count: number; results: object[] }>, string>
```
Comment (jonkoops) [inline src/graphql/config.ts:8]: Do we not have a better type than `object[]`?

File: src/graphql/index.ts [hunk 4]
```diff
@@ -23,6 +23,7 @@ export default graphqlHTTP((req) => ({
       cms: createDataLoader(''),
       data: createDataLoader(req.headers.authorization || ''),
       datasets: createDataLoader(''),
+      openAPI: createDataLoader(''),
```
Comment (jonkoops) [inline src/graphql/index.ts:26]: A whole new dataloader? How come this was missing?

**Review by jonkoops — APPROVED** [review]

File: src/fixtures/openapi/types.ts (hunk 1 above)
Comment (chrisvanmook) [reply src/fixtures/openapi/types.ts:1]: Exactly!

File: src/generated/drupal.ts (hunk 2 above)
Comment (chrisvanmook) [reply src/generated/drupal.ts]: I've absolutely no idea, I reverted this now. Going to look into it to fully understand how this is generated

File: src/graphql/index.ts (hunk 4 above)
Comment (chrisvanmook) [reply src/graphql/index.ts:26]: Well we used to use the `datasets`  loader to get results from openAPI in the dataset resolver. However, the results from openAPI are also used in filters resolver (thus using the datasets loader), so I thought it'd make more sense to create a separate loader for this.

Comment (jonkoops) [reply src/graphql/index.ts:26]: Cool, sounds good to me :+1:

File: src/graphql/config.ts (hunk 3 above)
Comment (chrisvanmook) [reply src/graphql/config.ts:8]: As soon as I will add the fixtures for data search API's in atlas, I will create a separate PR that will type this (or let's wait when the generic API Client is made 😄 )

**Pull request merged**